lib/OpenQA/Worker/Job.pm

```diff
@@ -0,0 +1,1263 @@
+            $interval,
+            sub {
+                $self->_upload_results;
+            });
+    }
+
+    $self->{_is_uploading_results} = 0;
+    $self->emit(uploading_results_concluded => {});
+    $callback->() if $callback;
+}
+
+sub post_upload_progress_to_liveviewhandler {
+    my ($self, $upload_up_to, $callback) = @_;
+
+    my $current_test_module = $self->current_test_module;
+    my %new_progress_info   = (
+        upload_up_to                => $upload_up_to,
+        upload_up_to_current_module => $current_test_module && $upload_up_to && $current_test_module eq $upload_up_to,
+        outstanding_files           => scalar(@{$self->files_to_send}),
+        outstanding_images          => scalar(%{$self->images_to_send}),
+    );
+
+    # skip if the progress hasn't changed
+    my $progress_changed;
+    my $current_progress_info = $self->progress_info;
+    for my $key (qw(upload_up_to upload_up_to_current_module outstanding_files outstanding_images)) {
+        my $new_value = $new_progress_info{$key};
+        my $old_value = $current_progress_info->{$key};
+        if (defined($new_value) != defined($old_value) || (defined($new_value) && $new_value ne $old_value)) {
+            $progress_changed = 1;
+            last;
+        }
+    }
+    if (!$progress_changed) {
+        return $callback->() if $callback;
+        return undef;
+    }
+    $self->{_progress_info} = \%new_progress_info;
+
+    my $job_id = $self->id;
+    $self->client->send(
+        post => "/liveviewhandler/api/v1/jobs/$job_id/upload_progress",
+        service_port_delta => 2,                     # liveviewhandler is supposed to run on web UI port + 2
+        json               => \%new_progress_info,
+        non_critical       => 1,
+        callback           => sub {
+            my ($res) = @_;
+            if (!$res) {
+                log_warning('Failed to post upload progress to liveviewhandler.');
+            }
+            $callback->($res);
+        });
+}
+
+sub upload_images {
+    my ($self) = @_;
+
+    my $tx;
+    my $job_id = $self->id;
+    my $client = $self->client;
+    my $ua     = $client->ua;
+    my $ua_url = $client->url->clone;
+    $ua_url->path("jobs/$job_id/artefact");
+
+    my $pooldir        = $self->worker->pool_directory;
+    my $fileprefix     = "$pooldir/testresults";
+    my $images_to_send = $self->images_to_send;
+    for my $md5 (keys %$images_to_send) {
+        my $file = $images_to_send->{$md5};
+        $self->_optimize_image("$fileprefix/$file");
+
+        log_debug("Uploading $file as $md5");
+        my %form = (
+            file => {
+                file     => "$fileprefix/$file",
+                filename => $file
+            },
+            image => 1,
+            thumb => 0,
+            md5   => $md5
+        );
+        # don't use api_call as it retries and does not allow form data
+        $tx = $ua->post($ua_url => form => \%form);
+        # FIXME: error handling?
```

Comment:
Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit or ticket